### DISCUSSION TITLE
Does calling connect() multiple times hurt performance in any way?

### DISCUSSION THREAD

**Discussion by danboyle8637** [planetscale/discussion#412] [FAQs]
I am building a `Class` with methods that control each query to the database. 

Right now I call the `connect` method at the beginner of each of my methods. Does calling this hurt in anyway? 

What is the connection is already open and a method calls the `connect` again... does it just ignore it?

Thanks!

**Comment by orware** [comment]
Hello @danboyle8637,

I've been meaning to circle back to responding to your question above here so I apologize for the extended wait time and I'm sure you've likely already had an opportunity to run some tests related to your question.

Overall, based on how I'm interpreting the [database-js](https://github.com/planetscale/database-js) code I would expect `connect()` to return an entirely new `Connection()` object each time it's called, which would mean that the `createSession()` method would also implicitly need to be run the first time a query is executed for that connection.

To me, this would likely lead to additional wait time as the new sessions would need to be established in each of these cases, rather than being able to reuse an existing session.

Your experiments since your post above hopefully may have confirmed something along these lines already, but I would say generally if you can create the connection once and then reuse that for multiple queries that may need to be executed on the page, that may be best overall. 

However, I would be curious to know if your experience doesn't confirm the behavior expectation above I shared.
